plugins/modules/files/iso_customize.py

```diff
@@ -0,0 +1,344 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright (c) 2022, Ansible Project
+# Copyright (c) 2022, VMware, Inc. All Rights Reserved.
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+DOCUMENTATION = r'''
+---
+module: iso_customize
+short_description: Customize the ISO file by add/delete/modify files
+description:
+  - This module is used to customize the ISO file by add/delete files.
+author:
+  - Yuhua Zou (@ZouYuhua) <zouy@vmware.com>
+requirements:
+  - "pycdlib"
+  - "python >= 2.7"
+version_added: '5.6.0'
+
+options:
+  src_iso:
+    description:
+    - This is the path of source ISO file.
+    - Will fail if specified ISO file does not exist on local machine.
+    - 'Note: With all ISO9660 levels from 1 to 3, all file names are restricted to uppercase letters, numbers and
+      underscores (_). File names are limited to 31 characters, directory nesting is limited to 8 levels, and path
+      names are limited to 255 characters.'
+    type: str
+    required: true
+  dest_iso:
+    description:
+    - The path with file name of the customized ISO file on local machine.
+    type: str
+    required: true
+  add_files:
+    description:
+    - Allows to add and replace files in the ISO file.
+    - Will create intermediate folders inside the ISO file when they do not exist.
+    type: list
+    required: false
+    elements: dict
+    suboptions:
+      src_file:
+        description:
+        - The path with file name on the machine the module is executed on.
+        type: path
+      dest_file:
+        description:
+        - The absolute path of the file inside the ISO file.
+        type: str
+  delete_files:
+    description:
+    - Absolute paths for files inside the ISO file that should be removed.
+    type: list
+    required: false
+    elements: str
+'''
+
+EXAMPLES = r'''
+- name: "Customize ISO file"
+  community.general.iso_customize:
+    src_iso: "/path/to/ubuntu-22.04-desktop-amd64.iso"
+    dest_iso: "/path/to/ubuntu-22.04-desktop-amd64-customized.iso"
+    delete_files:
+      - "/preseed/ubuntu.seed"
+    add_files:
+      - src_file: "/path/to/grub.cfg"
+        dest_file: "/boot/grub/grub.cfg"
+      - src_file: "/path/to/ubuntu.seed"
+        dest_file: "/preseed/ubuntu.seed"
```

Comment:
In this example, what is the outcome for the path `/preseed/ubuntu.seed` ? Will the resulting ISO file have it deleted or will it have the added file?

There is no mention in the documentation about what happens in such case.

Personally, I think it would be better to have a `state` parameter with choices `present` and `absent`, like many other modules, but that is a personal preference.

Reply:
Add comment in the documentation for such case:
description:
- In the example, the file /preseed/ubuntu.seed is deleted at first. Then we add local file
- /path/to/ubuntu.seed to /path/to/ubuntu.seed inside ISO. So the file /path/to/ubuntu.seed
- inside ISO is replaced.

Thanks

Reply:
These documentation blobs are used to generate the HTML docs, as in (example) :

https://docs.ansible.com/ansible/latest/collections/community/general/xfconf_info_module.html

This YAML comment you have added will not be included in the output for the end user. 

And I placed the comment on the EXAMPLES block, but you probably want to write about that in DOCUMENTATION.

Reply:
Thanks. Move the comment to section 'description' of DOCUMENTATION. 
Not sure if it will meet your expection.

Reply:
@russoz 
Please kindly help check the if the changes meet your expection or not. Thanks

Reply:
I would make it explicit to the end user that execution order is: all `delete` actions first, then all `add` actions.

Keep in mind that the end user will NOT look into the Python code. Many of them will not even know Python.

Reply:
> I would make it explicit to the end user that execution order is: all `delete` actions first, then all `add` actions.
> 
> Keep in mind that the end user will NOT look into the Python code. Many of them will not even know Python.

Done. Thanks
I mark this issue as "Resolved" in order to remove label "needs_revision"

Please reopen it if it's not meet your expection.   Thanks.

Reply:
> I would make it explicit to the end user that execution order is: all `delete` actions first, then all `add` actions.
> 
> Keep in mind that the end user will NOT look into the Python code. Many of them will not even know Python.

Fix it. Please help review the document part.  Thanks